packages/datadog-plugin-aws-sdk/src/services/bedrockruntime.js

```diff
@@ -0,0 +1,254 @@
+    return {
+      prompt: requestBody.texts,
+      input_type: requestBody.input_type || '',
+      truncate: requestBody.truncate || ''
+    }
+  } else if (provider === COHERE) {
+    return {
+      prompt: requestBody.prompt,
+      temperature: requestBody.temperature || '',
+      top_p: requestBody.p || '',
+      top_k: requestBody.k || '',
+      max_tokens: requestBody.max_tokens || '',
+      stop_sequences: requestBody.stop_sequences || [],
+      stream: requestBody.stream || '',
+      n: requestBody.num_generations || ''
+    }
+  } else if (provider === META) {
+    return {
+      prompt: requestBody.prompt,
+      temperature: requestBody.temperature || '',
+      top_p: requestBody.top_p || '',
+      max_tokens: requestBody.max_gen_len || ''
+    }
+  } else if (provider === MISTRAL) {
+    return {
+      prompt: requestBody.prompt,
+      max_tokens: requestBody.max_tokens || '',
+      stop_sequences: requestBody.stop || [],
+      temperature: requestBody.temperature || '',
+      top_p: requestBody.top_p || '',
+      top_k: requestBody.top_k || ''
+    }
+  } else if (provider === STABILITY) {
+    return {}
+  }
+  return {}
+}
+
+function extractTextAndResponseReason (response, provider, modelName, shouldSetChoiceIds) {
+  const body = JSON.parse(Buffer.from(response.body).toString('utf8'))
+  const textAndResponseReasons = []
+
+  try {
+    if (provider === AI21) {
+      if (modelName.includes('jamba')) {
+        const generations = body.choices || []
+        generations.forEach(generation => {
+          textAndResponseReasons.push({
+            text: generation.message || '',
+            finish_reason: generation.finish_reason || '',
+            choice_id: shouldSetChoiceIds ? generation.id : undefined
+          })
+        })
+      }
+      const completions = body.completions || []
+      completions.forEach(completion => {
+        textAndResponseReasons.push({
+          text: completion.data?.text || '',
+          finish_reason: completion.finishReason || '',
+          choice_id: shouldSetChoiceIds ? completion.id : undefined
+        })
+      })
+    } else if (provider === AMAZON && modelName.includes('embed')) {
+      textAndResponseReasons.push({
+        text: body.embedding || [],
+        finish_reason: '',
+        choice_id: undefined
+      })
+    } else if (provider === AMAZON) {
+      (body.results || []).forEach(result => {
+        textAndResponseReasons.push({
+          text: result.outputText || '',
+          finish_reason: result.completionReason || '',
+          choice_id: undefined
+        })
+      })
+    } else if (provider === ANTHROPIC) {
+      textAndResponseReasons.push({
+        text: body.completion || body.content || '',
+        finish_reason: body.stop_reason || '',
+        choice_id: undefined
+      })
+    } else if (provider === COHERE && modelName.includes('embed')) {
+      (body.embeddings || [[]]).forEach(embedding => {
```

Comment:
Maybe it's intentional? I suppose line 218 also adds an empty entry, albeit with `text` being an empty string instead of an empty array.

In either case it might be beneficial to add a comment above `const textAndResponseReasons` to explain what goes in it and if it's fine to have empty entries or not in it.

Reply:
Yes it is intentional as the Cohere model has a response struct set as a list of embeddings https://docs.aws.amazon.com/bedrock/latest/userguide/model-parameters-embed.html#model-parameters-embed-request-response

Reply:
I added more clarification in the recent refactoring commits that would stringify the fields if they are not strings.
Thanks for noting this.